src/libsync/abstractnetworkjob.cpp

```diff
@@ -173,7 +173,7 @@ void AbstractNetworkJob::slotFinished()
     // get the Date timestamp from reply
     _responseTimestamp = _reply->rawHeader("Date");
 
-    if (_followRedirects) {
+    if (_followRedirects && reply()->operation() == QNetworkAccessManager::GetOperation) {
```

Comment:
@vladimiroff It would be better to check after `_redirectCount` was incremented, log something and then not trigger a follow-up request (similar to the other error cases)